app/Http/Controllers/ImageController.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Resources\CategoryResource;
+use App\Models\Category;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use App\Http\Requests\CategoryRequest;
+
+class ImageController extends Controller
+{
+    public function uploadIcon(CategoryRequest $request)
```

Comment:
@Geri-PG This should be the endpoint for uploading the category icon. To validate the request you use CategoryRequest, which will return 422 if the required parameters for creating a new category are not provided.

Reply:
Stavio je Geri.

Reply:
Je li ispravio na nekom drugom tiketu? Jer mi ovdje i dalje prikazuje da koristi CategoryRequest koji zahtijeva name i description parametre tako da ce vratiti 422 gresku

Reply:
@VoxMne Mozes dodati u f-ji parametar Category $category jer si vec definisao tako rutu, pa ti onda ne treba ovaj dio u kodu:
    $categoryId = $request->input('category_id');
    $category = Category::findOrFail($categoryId);
Koristeci route model binding laravel u pozadini vozi racuna da li postoji rekord ili ne, ukoliko ne pronadje kategoriju vrace 404. Takodje umjesto CategoryRequest koristi novi Request za uploadIcon zato sto ce ti CategoryRequest vratiti 422 jer ocekuje i ostale parametre.